src/App.tsx

```diff
@@ -1,14 +1,62 @@
-/* eslint-disable max-len */
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
-import { TodoList } from './components/TodoList';
+import { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api';
+import { Status } from './types/Status';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoList } from './components/TodoList';
+import { TodoModal } from './components/TodoModal';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as todosActions } from './features/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './features/filter';
 
 export const App: React.FC = () => {
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const { query, status }: Filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  const [loading, setLoading] = useState(true);
+
+  const setTodos = (newTodos: Todo[]) => dispatch(todosActions.set(newTodos));
+
+  useEffect(() => {
+    setTimeout(() => {
+      setLoading(false);
+    }, 500);
+
+    getTodos().then(setTodos);
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    setTimeout(() => {
      setLoading(false);
    }, 500);

    getTodos().then(serverTodos=> dispatch(todosActions.set(serverTodos));
  }, []);
```
you don't need this function